### PR TITLE
OpenTitan: Application fixups

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

There have been some recent regressions to running applications on OpenTitan. This PR fixes those.

### Testing Strategy

Running usersapce apps on OT.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: arch/rv32i/src/epmp.rs [hunk 1]
```diff
@@ -667,11 +667,17 @@ impl<const MAX_AVAILABLE_REGIONS_OVER_TWO: usize> kernel::platform::mpu::MPU
...
                                 | (csr::CSR.pmpconfig_get(x / 2) & other_region_mask),
                         );
+
                         csr::CSR.pmpaddr_set(x * 2, (start) >> 2);
                         csr::CSR.pmpaddr_set((x * 2) + 1, (start + size) >> 2);
+
+                        csr::CSR.pmpconfig_set(
+                            x / 2,
+                            (cfg_val << 8) << region_shift | (csr::CSR.pmpconfig_get(x / 2)),
+                        );
```
Comment (hudson-ayers) [inline arch/rv32i/src/epmp.rs:680]: did I break something here in https://github.com/tock/tock/pull/2757 ?

Comment (alistair23) [reply arch/rv32i/src/epmp.rs:680]: Yep, for me apps stopped working when using ePMP

Comment (alistair23) [reply arch/rv32i/src/epmp.rs:680]: The problem is that we could have junk left over in the address register. As the high priority PMP configs are used for the apps they will take effect before the kernel configs (which are the lowest priority).

We have disabled the pmp regions in `disable_app_mpu()`. If we enable them before writing the address we can lock the kernel out of execute permissions. So we have to make sure we set the address first

Comment (alistair23) [reply arch/rv32i/src/epmp.rs]: I added more comments here

**Review by bradjc — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]

File: arch/rv32i/src/epmp.rs (hunk 1 above)
Comment (hudson-ayers) [reply arch/rv32i/src/epmp.rs]: Ah, I understand. I wanted to make sure this breakage would not also apply for the normal pmp file, sounds like it should not. Thanks for the explanation!

**Review by ppannuto — COMMENTED** [review]
bors r+

**Pull request merged**